app/models/iteration.rb

```diff
@@ -138,6 +93,34 @@ def self.task_progress
     progress_hash
   end
 
+  # Tasks for each project's current iteration
+  def self.collect_current_tasks
+    tasks_iter = {}
+    Project.all.each do |proj|
+      iter = Iteration.current_iter proj
+      tasks_iter[proj.id] = []
+      if iter != nil
+        tasks = Task.where(iteration_id: iter.id)
+        tasks_iter[proj.id] = tasks
+      end
+    end
+    tasks_iter
+  end
+
+  # return true if the time stamp is inside the time
+  def current_iter? time
+    self.start_time < time and self.end_time > time
+  end
+
+  # return the current processing iteration for project
+  def self.current_iter proj
+    iters = Iteration.where(project_id: proj.id).where("start_time<?",Time.now).where("end_time>?",Time.now)
```

Comment:
You can use active record's support for filtering by date.

See this post: https://stackoverflow.com/questions/2381718/rails-activerecord-date-between

Using the native support removes the dependency on db types.

Reply:
I wonder if the case of the stack overflow is applicable here though. Since I'm returning the instance according to the date assigned before